subgraphs/axelar/src/mappings.ts

```diff
@@ -0,0 +1,762 @@
+  const dstPoolId = dstNetworkConstants.getPoolAddress();
+
+  const dstStr = event.params.destinationAddress;
+  const dstAddress = isValidEVMAddress(dstStr)
+    ? Address.fromString(dstStr)
+    : Bytes.fromUTF8(dstStr);
+
+  log.info("[handleTokenSent]original={},dstAddress={}", [
+    event.params.destinationAddress,
+    dstAddress.toHexString(),
+  ]);
+
+  const customEvent = _createCustomEvent(event)!;
+  _handleTransferOut(
+    Address.fromString(tokenAddress),
+    event.params.sender,
+    dstAddress,
+    event.params.amount,
+    dstChainId,
+    poolId,
+    dstPoolId,
+    BridgePoolType.BURN_MINT,
+    CrosschainTokenType.WRAPPED,
+    customEvent,
+    null
+  );
+}
+
+export function handleContractCallWithToken(
+  event: ContractCallWithToken
+): void {
+  const tokenSymbol = getOrCreateTokenSymbol(event.params.symbol);
+  const tokenAddress = tokenSymbol.tokenAddress;
+  const poolId = event.address.concat(Address.fromString(tokenAddress));
+  const dstChainId = networkToChainID(
+    event.params.destinationChain.toUpperCase()
+  );
+  const dstNetworkConstants = getNetworkSpecificConstant(dstChainId);
+  const dstPoolId = dstNetworkConstants.getPoolAddress();
+
+  const dstStr = event.params.destinationContractAddress;
+  const dstAccount = isValidEVMAddress(dstStr)
+    ? Address.fromString(dstStr)
+    : Bytes.fromUTF8(dstStr);
+
+  const customEvent = _createCustomEvent(event)!;
+
+  const account = _handleTransferOut(
+    Address.fromString(tokenAddress),
+    event.params.sender,
+    dstAccount,
+    event.params.amount,
+    dstChainId,
+    poolId,
+    dstPoolId,
+    BridgePoolType.BURN_MINT,
+    CrosschainTokenType.WRAPPED,
+    customEvent,
+    null
+  );
+
+  account.messageOut(dstChainId, dstAccount, event.params.payload);
+}
+
+export function handleContractCall(event: ContractCall): void {
+  // contract call
+  const dstChainId = networkToChainID(
+    event.params.destinationChain.toUpperCase()
+  );
+  const dstAccount = Address.fromString(
+    event.params.destinationContractAddress
+  );
+
+  const customEvent = _createCustomEvent(event)!;
+  _handleMessageOut(
+    dstChainId,
+    event.params.sender,
+    dstAccount,
+    event.params.payloadHash,
+    customEvent
+  );
+}
+
+export function handleContractCallApproved(event: ContractCallApproved): void {
+  // contract call
+  const srcChainId = networkToChainID(event.params.sourceChain.toUpperCase());
+  const srcAccount = Address.fromString(event.params.sourceAddress);
+  const customEvent = _createCustomEvent(event)!;
+  _handleMessageIn(
+    srcChainId,
+    srcAccount,
+    event.address,
+    event.params.payloadHash,
+    customEvent
+  );
+}
+
+export function handleContractCallApprovedWithMint(
+  event: ContractCallApprovedWithMint
+): void {
+  // contract call
+  const srcChainId = networkToChainID(event.params.sourceChain.toUpperCase());
+  // this is needed to support transferIn from non-EVM chains
```

Comment:
I don't think any of the chains are non evm. Do you mean non-ethereum chains?

Reply:
There are transfers in/out with cosmos chain, and they have non-evm addresses.

Reply:
Ah, gotcha